Assets/Scripts/StartupController.cs

```diff
@@ -0,0 +1,113 @@
+	void Start () {
+		//PlayerPrefs.DeleteKey("session"); // Uncomment this to test from login screen
+		Server.Connect();
+		// If session token works, go to game, otherwise remove token and init login UI
+		if(PlayerPrefs.HasKey("session")) {
+			if(!Server.RetryLogin()) {
+				PlayerPrefs.DeleteKey("session");
+				InitLoginUI();
+			}else{
+				GoToMain();
+			}
+		}else{
+			InitLoginUI();
+		}
+	}
+
+	// Runs every frame, lerps objects if _expanding or _collapsing
+	void Update() {
+		if(_t >= 1f) {
+			_expanding = false;
+			_collapsing = false;
+			_t = 0f;
+		}else if(_expanding) {
+			_t += Time.deltaTime;
+			_panelRT.sizeDelta = new Vector2(150f,Mathf.Lerp(_panelRT.sizeDelta.y,260f,_t));
+			_emailRT.anchoredPosition = new Vector3(0,Mathf.Lerp(_emailRT.anchoredPosition.y,-148.75f,_t));
+			_passwordRT.anchoredPosition = new Vector3(0,Mathf.Lerp(_passwordRT.anchoredPosition.y,-178.75f,_t));
+			_confirmPasswordRT.anchoredPosition = new Vector3(0,Mathf.Lerp(_confirmPasswordRT.anchoredPosition.y,-208.75f,_t));
+		}else if(_collapsing) {
+			_t += Time.deltaTime;
+			_panelRT.sizeDelta = new Vector2(150f,Mathf.Lerp(_panelRT.sizeDelta.y,200f,_t));
+			_emailRT.anchoredPosition = new Vector3(0,Mathf.Lerp(_emailRT.anchoredPosition.y,-118.75f,_t));
+			_passwordRT.anchoredPosition = new Vector3(0,Mathf.Lerp(_passwordRT.anchoredPosition.y,-148.75f,_t));
+			_confirmPasswordRT.anchoredPosition = new Vector3(0,Mathf.Lerp(_confirmPasswordRT.anchoredPosition.y,-148.75f,_t));
+		}
+	}
+
+	// Sets vars when login or register is clicked. Public so that it can be called from the scene
+	public void ToggleLoginRegister(string buttonText) {
+		if(selectedButton == "Login" && buttonText == "Register") {
+			selectedButton = "Register";
+			_t = 0f;
+			_collapsing = false;
+			_expanding = true;
+		}else if(selectedButton == "Register" && buttonText == "Login") {
+			selectedButton = "Login";
+			_t = 0f;
+			_expanding = false;
+			_collapsing = true;
+		}
+	}
+
+	// Inits vars for login UI
+	private void InitLoginUI() {
+		selectedButton = "Login";
+		_expanding = false;
+		_collapsing = false;
+		_t = 0f;
+
+		_panelRT = (RectTransform)GameObject.Find("Panel").transform;
+		_emailRT = (RectTransform)GameObject.Find("Email").transform;
+		_passwordRT = (RectTransform)GameObject.Find("Password").transform;
+		_confirmPasswordRT =(RectTransform)GameObject.Find("ConfirmPassword").transform;
+	}
+
+	// Attempt to register or login. If successful, go to game. Public so can be called from scene
+	public void Login() {
+		string username = GameObject.Find("Username").GetComponent<InputField>().text;
+		string password = GameObject.Find("Password").GetComponent<InputField>().text;
+
+		if(selectedButton == "Register") {
+			string email = GameObject.Find("Email").GetComponent<InputField>().text;
+			string confirmpw = GameObject.Find("ConfirmPassword").GetComponent<InputField>().text;
+			if(string.Equals(password, confirmpw)) {
```

Comment:
Okay.  I'm going to approve, but I do have a suggestion.  Just styling, so you could ignore it, but the extra level of if statement seems unnecessary, so instead of if( if()else)else()
if(passmatch)
else()
if(create user success)
else()